libsolidity/formal/SMTEncoder.cpp

```diff
@@ -3162,11 +3162,11 @@ void SMTEncoder::collectFreeFunctions(set<SourceUnit const*, ASTNode::CompareByI
 				auto contract = dynamic_cast<ContractDefinition const*>(node.get());
 				contract && contract->isLibrary()
 			)
-			{
 				for (auto function: contract->definedFunctions())
-					if (!function->isPublic())
+					// We need to add public library functions too because they can be called
+					// internally by internal library functions that are considered free functions.
```

Comment:
Why are internal library functions considered free functions?

Reply:
Because they are called in the context of the caller contract, so for the encoding this has the same effect.